core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsService.java

```diff
@@ -0,0 +1,113 @@
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.concurrent.TimeUnit;
+
+public class NlpjsService {
+
+    private final static String NLPJS_BASE_PATH = "/api";
+
+    private String nlpjsServer;
+
+    private NlpjsApi nlpjsApi;
+
+    private Gson gson;
+
+    public NlpjsService(@NonNull String nlpjsServer) {
+        this.nlpjsServer = nlpjsServer;
+        gson = new GsonBuilder().registerTypeAdapter(Resolution.class, new ResolutionDeserializer())
+                .create();
+        String nlpjsApiFullPath = this.nlpjsServer + NLPJS_BASE_PATH + "/";
+        OkHttpClient httpClient = new OkHttpClient.Builder()
+                .addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
+                .readTimeout(3, TimeUnit.SECONDS)
+                .build();
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(nlpjsApiFullPath)
+                .client(httpClient)
+                .addConverterFactory(GsonConverterFactory.create(gson)).build();
+        this.nlpjsApi = retrofit.create(NlpjsApi.class);
+    }
+
+
+    public Agent getAgentInfo(String agentId) throws IOException, IntentRecognitionProviderException {
+        Response<Agent> agentResponse = nlpjsApi.getAgentInfo(agentId).execute();
+        if (!agentResponse.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(agentResponse.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error message \"{3}\" ",
+                    nlpjsApi.getAgentInfo(agentId).request().method(),
+                    nlpjsApi.getAgentInfo(agentId).request().url(), agentResponse.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
+        }
+        return agentResponse.body();
+    }
+
+
+    public boolean createAgent(AgentInit agentInit) throws IOException, IntentRecognitionProviderException {
+        Response<Void> response = nlpjsApi.createAgent(agentInit).execute();
+        if (!response.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(response.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error message \"{3}\" ",
+                    nlpjsApi.createAgent(agentInit).request().method(),
+                    nlpjsApi.createAgent(agentInit).request().url(), response.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
```

Comment:
Just throw the exception, no need to log, this should be propagated to a catch that takes care of logging the error (let me know if it is not the case)

Reply:
👍